astropy/config/paths.py

```diff
@@ -187,7 +187,11 @@ def __init__(self, path=None, delete=False):
 
     def __enter__(self):
         self.__class__._temp_path = self._path
-        return self._default_path_getter('astropy')
+        try:
```

Comment:
If this goes into 4.0.1 instead of 4.0, this change needs a bug fix change log because this line was introduced in #8237 . cc @Cadair

Reply:
this will land in 4.0 as it's almost ready and we're still waiting for upstream blockers.